community/values/src/main/java/org/neo4j/values/storable/TemporalValue.java

```diff
@@ -1366,6 +1367,32 @@ static int validNano( AnyValue millisecond, AnyValue microsecond, AnyValue nanos
         return (int) (ms * 1000_000 + us * 1000 + ns);
     }
 
+    static <TEMP extends Temporal> TEMP withTruncatedNano( TEMP value )
+    {
+        int precision = Values.getNanoPrecision();
+        if ( precision < 0 || precision > 9 )
+        {
+            throw new InvalidValuesArgumentException( "The dbms.temporal.nanosecond_precision config " +
+                    "must be an integer between 0 and 9, was " + precision );
+        }
+        if ( precision == 9 )
+        {
+            return value;
+        }
+        int truncateFactor = (int) Math.pow( 10, 9 - precision );
+        TEMP newValue;
+        try
+        {
+            int truncatedNano = value.get( NANO_OF_SECOND ) / truncateFactor * truncateFactor;
+            newValue = (TEMP) value.with( NANO_OF_SECOND, truncatedNano );
+        }
+        catch ( DateTimeException | ArithmeticException e )
```

Comment:
Can this catch block be removed? `DateTimeException` and `ArithmeticException` are runtime exceptions anyway